tutoraspects/templates/aspects/apps/aspects/scripts/dbt.sh

```diff
@@ -26,11 +26,28 @@ cd aspects-dbt || exit
 
 export ASPECTS_EVENT_SINK_DATABASE={{ASPECTS_EVENT_SINK_DATABASE}}
 export ASPECTS_XAPI_DATABASE={{ASPECTS_XAPI_DATABASE}}
+export DBT_STATE={{ DBT_STATE_DIR }}
 
 echo "Installing dbt dependencies"
 dbt deps --profiles-dir /app/aspects/dbt/
 
-echo "Running dbt $*"
-dbt "$@" --profiles-dir /app/aspects/dbt/
-
+echo "Running dbt ${@:2}"
+
+if [ "$1" == "True" ]
+then
+  echo "Requested to only run modified state, checking for ${DBT_STATE}/manifest.json"
+fi
+
+# If state exists and we've asked to only run changed files, add the flag
+if [ "$1" == "True" ] && [ -e "${DBT_STATE}/manifest.json" ]
+then
+  echo "Found {{DBT_STATE_DIR}}/manifest.json so only running modified items and their downstreams"
+  dbt "${@:2}" --profiles-dir /app/aspects/dbt/ -s state:modified+
```

Comment:
The dbt docs aren't clear on this, but I think we also need to specify where dbt should look for the old manifest by passing `--state ${DBT_STATE}` to the dbt command.

Reply:
The `DBT_STATE` env var works too, so I'm using that above to keep the command from getting ridiculous